stdlib/public/core/StringWordBreaking.swift

```diff
@@ -0,0 +1,690 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension _StringGuts {
+  @inline(__always)
+  internal func roundDownToNearestWord(
+    _ i: String._WordView.Index
+  ) -> String._WordView.Index {
+    _internalInvariant(i._encodedOffset <= count)
+
+    let offset = i._encodedOffset
+
+    if offset == 0 || offset == count {
+      return i
+    }
```

Comment:
Are you expecting this to be a super common early exit that's often known at compile time of the user's code? Otherwise probably drop the inline always unless there's a need

Reply:
The `roundDownToNearestWord`/`_slowRoundDownToNearestWord` split is left over from the version where we had performance flags for word boundaries in `String.Index`. Now that we don't have them, this scheme doesn't seem particularly useful, as we'll almost always fall into the "slow" case.